crates/rooch-framework/sources/coin_store.move

```diff
@@ -125,52 +161,72 @@ module rooch_framework::coin_store {
         deposit_internal(coin_store_obj, coin)
     }
 
-    // We do not allow to transfer a CoinStore to another account, CoinStore is default ownerd by the system.
-    // Only provide a internal function for account_coin_store.
-    public fun transfer<CoinType: key>(coin_store_obj: Object<CoinStore<CoinType>>, owner: address){
-        // Cannot transfer a frozen CoinStore
-        // We do not use the frozen Object to represent the frozen CoinStore, because we want allow the T module to unfreeze the CoinStore
-        assert!(!object::borrow(&coin_store_obj).frozen, error::permission_denied(ErrorCoinStoreIsFrozen));
-        //TODO if the owner has the CoinStore<CoinType>, we should merge this CoinStore<CoinType> to the owner's CoinStore<CoinType>
-        //So, we need to migrate account_coin_store to coin_store module.
-        object::transfer_extend(coin_store_obj, owner)
+    // We do not allow to transfer a CoinStore to another account, this function will abort directly.
+    // Because we need to ensure one Account only has one CoinStore for one CoinType
+    // If you want tranfer a CoinStore to another account, you can call `coin_store::remove(Object<CoinStore<CoinType>>)` and deposit the Coin<CoinType> to another account.
+    public fun transfer<CoinType: key>(_coin_store_obj: Object<CoinStore<CoinType>>, _owner: address){
+        abort error::not_implemented(ErrorCoinStoreTransferNotSupported)
```

Comment:
If we do not allow the transfer of a CoinStore, would it be better to remove the method than raise an error directly?

Reply:
1. The Object without `store` should provide a public transfer function like the `ERC20` standard.
2. For comment.